plugins/inputs/apcupsd/apcupsd.go

```diff
@@ -0,0 +1,75 @@
+package apcupsd
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/mdlayher/apcupsd"
+)
+
+const defaultAddress = "127.0.0.1:3551"
+
+type ApcUpsd struct {
+	Servers []string
+}
+
+func (_ *ApcUpsd) Description() string {
+	return "Monitor APC UPSes connected to apcupsd"
+}
+
+var sampleConfig = `
+  # a list of running apcupsd server to connect to. 
+  # If not provided will default to 127.0.0.1:3551
+  servers = ["127.0.0.1:3551"]
+`
+
+func (_ *ApcUpsd) SampleConfig() string {
+	return sampleConfig
+}
+
+func (h *ApcUpsd) Gather(acc telegraf.Accumulator) error {
+	for _, addr := range h.Servers {
+		status, err := fetchStatus(addr)
+		if err != nil {
+			return err
+		}
+
+		tags := map[string]string{
+			"serial":   status.SerialNumber,
+			"ups_name": status.UPSName,
+		}
+
+		fields := map[string]interface{}{
+			"online":                 status.Status == "ONLINE",
```

Comment:
Maybe we should have a tag for status?  We can still have this information encoded as a field but we should consider encoding it as an int instead of a bool, which while harder to understand does work better with many visualization programs.